lib/chai/core/assertions.js

```diff
@@ -528,10 +528,25 @@ module.exports = function (chai, _) {
    */
 
   Assertion.addProperty('empty', function () {
-    var obj = flag(this, 'object');
-    new Assertion(obj).to.exist;
+    var val = flag(this, 'object')
+      , itemsCount;
+
+    switch (_.type(val)) {
+      case 'array':
+      case 'string':
+        itemsCount = val.length;
+        break;
+      case 'function':
+        throw new TypeError('.empty() was passed a function');
+      default:
+        if (val !== Object(val)) {
+          throw new TypeError('.empty() was passed non-string primitive');
+        }
+        itemsCount = Object.keys(val).length;
+    }
+
     this.assert(
-        Object.keys(Object(obj)).length === 0
+        0 === itemsCount
```

Comment:
@shvaikalesh Yes, we thought about that too, we already discussed that [here](https://github.com/chaijs/chai/issues/737#issuecomment-229646757) and we came to the conclusion that by moving all our code to separate repos with their own linting rules would be better, however, if you disagree you're welcome to share your thoughts with us 😄 .